bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1116,6 +1143,38 @@ private void setThingStatus(Thing thing, ThingStatusInfo thingStatusInfo) {
         }
     }
 
+    @Override
+    public void setEnabled(ThingUID thingUID, boolean enabled) {
+        Thing thing = getThing(thingUID);
+
+        if (enabled) {
+            // Enable a thing
+            logger.debug("Thing {} will be enabled.", thing.getUID().getAsString());
+            if (storage != null) {
+                storage.remove(thingUID.getAsString());
+            }
+            if (!ThingHandlerHelper.isHandlerInitialized(thing)) {
+                setThingStatus(thing, buildStatusInfo(ThingStatus.UNINITIALIZED, ThingStatusDetail.NONE));
+                registerAndInitializeHandler(thing, findThingHandlerFactory(thing.getThingTypeUID()));
+            }
+        } else {
+            logger.debug("Thing {} will be disabled.", thing.getUID().getAsString());
+            if (storage != null) {
+                storage.put(thingUID.getAsString(), enabled);
+            }
+            ThingHandlerFactory thingHandlerFactory = findThingHandlerFactory(thing.getThingTypeUID());
+            if (isHandlerRegistered(thing)) {
```

Comment:
You wrote the new enablement status into the storage and call `unregisterAndDisposeHandler(...)` below, if a handler exists. You rely on `doUnregisterHandler(...)` to set the thing to `UNINITIALIZED`/`DISABLED`. 

However, what if there is not handler, e.g. because of a problem leading to `UNINITIALIZED`/`HANDLER_MISSING_ERROR`? Then suddenly the disabled-state will not be reflected anywhere.

Reply:
It will not be reflected in the thing's status, but the storage will keep the information for that thing. Later when `registerAndInitializeHandler(...)` is called to register handler for that thing the information in the storage will be taken in consideration.